tools/prom-metrics-collector/metrics_json_generator.go

```diff
@@ -3,17 +3,63 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"os"
+	"strings"
+
+	"github.com/kubevirt/monitoring/pkg/metrics/parser"
+
+	sspMetrics "kubevirt.io/ssp-operator/pkg/monitoring/metrics/ssp-operator"
+	validatorMetrics "kubevirt.io/ssp-operator/pkg/monitoring/metrics/template-validator"
+	"kubevirt.io/ssp-operator/pkg/monitoring/rules"
 )
 
+// This should be used only for very rare cases where the naming conventions that are explained in the best practices:
+// https://sdk.operatorframework.io/docs/best-practices/observability-best-practices/#metrics-guidelines
+// should be ignored.
+var excludedMetrics = map[string]struct{}{}
+
 func main() {
-	metricFamilies := readMetrics()
+	err := sspMetrics.SetupMetrics()
+	if err != nil {
+		panic(err)
+	}
+
+	err = validatorMetrics.SetupMetrics()
+	if err != nil {
+		panic(err)
+	}
+
+	if err := rules.SetupRules(); err != nil {
+		panic(err)
+	}
+
+	var metricFamilies []parser.Metric
+
+	metricsList := sspMetrics.ListMetrics()
+	for _, m := range metricsList {
+		if _, isExcludedMetric := excludedMetrics[m.GetOpts().Name]; !isExcludedMetric {
+			metricFamilies = append(metricFamilies, parser.Metric{
+				Name: m.GetOpts().Name,
+				Help: m.GetOpts().Help,
+				Type: strings.ToUpper(string(m.GetBaseType())),
+			})
+		}
+	}
+
+	rulesList := rules.ListRecordingRules()
+	for _, r := range rulesList {
+		if _, isExcludedMetric := excludedMetrics[r.GetOpts().Name]; !isExcludedMetric {
+			metricFamilies = append(metricFamilies, parser.Metric{
+				Name: r.GetOpts().Name,
+				Help: r.GetOpts().Help,
+				Type: strings.ToUpper(string(r.GetType())),
+			})
+		}
+	}
 
 	jsonBytes, err := json.Marshal(metricFamilies)
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		panic(err)
 	}
 
-	fmt.Println(string(jsonBytes))
+	fmt.Println(string(jsonBytes)) // Write the JSON string to standard output
```

Comment:
Is that comment necessary? IMO it's obvious.

Reply:
removed